homeassistant/components/nws/weather.py

```diff
@@ -165,85 +68,62 @@ def convert_condition(time, weather):
     return cond, max(prec_probs)
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info):
     """Set up the NWS weather platform."""
-
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
     longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
-    station = config.get(CONF_STATION)
-    api_key = config[CONF_API_KEY]
-    mode = config[CONF_MODE]
-
-    websession = async_get_clientsession(hass)
-    # ID request as being from HA, pynws prepends the api_key in addition
-    api_key_ha = f"{api_key} homeassistant"
-    nws = SimpleNWS(latitude, longitude, api_key_ha, websession)
-
-    _LOGGER.debug("Setting up station: %s", station)
-    try:
-        await nws.set_station(station)
-    except ERRORS as status:
-        _LOGGER.error(
-            "Error getting station list for %s: %s", (latitude, longitude), status
-        )
-        raise PlatformNotReady
```

Comment:
Removing `PlatformNotReady` means this integration will never retry if nws is unreachable without a restart.

Consider moving this to a config entry via a config flow https://developers.home-assistant.io/docs/config_entries_config_flow_handler/ in a **new** PR after this one is done.

Reply:
This was the plan, and I think the reason why I had moved it out of the platform.  Would it be advisable to move this piece back here in the meantime?

Reply:
As long as you are going to get the follow up PR in time before 0.108 ships it can wait for the next PR

Reply:
I reverted this to happen inside `async_setup_platform`.  I can still try to have a PR for config flow by 0.108, but it is good to not have to rely on it.